### PR TITLE
Portere til kotlin

### PR DESCRIPTION
**Pull request by MengZhuNAV**
Port Dokgen to Kotlin. The code is mostly generated by IntelliJ but manual modifications are done to implement logics in a more Kotlin way. Configurations such as dependencies and plugins are also done manually.

The PR also port the build pipeline to Github actions and publish Dokgen Docker image to Github registry. There is another PR in familie-ba-dokgen which uses the image. After a few days of trying in familie-ba applications, we can notify other users about the new Docker image.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -128,6 +134,13 @@
             <version>1.13.1</version>
         </dependency>
 
+        <!-- https://mvnrepository.com/artifact/org.json/json -->
+        <dependency>
+            <groupId>org.json</groupId>
```
Comment (hensol) [inline pom.xml:139]: Hva er grunnen til at vi trenger denne?

File: pom.xml [hunk 2]
```diff
@@ -188,19 +218,33 @@
...
+            <plugin>
+                <groupId>org.jetbrains.kotlin</groupId>
+                <artifactId>kotlin-maven-plugin</artifactId>
+                <configuration>
+                    <args>
+                        <arg>-Xjsr305=strict</arg>
+                    </args>
+                    <compilerPlugins>
+                        <plugin>spring</plugin>
+                        <plugin>jpa</plugin>
```
Comment (hensol) [inline pom.xml:230]: Trenger vel ikke jpa her.

File: src/main/kotlin/no/nav/dokgen/configuration/DelayedShutdownHook.kt [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+import org.springframework.context.ConfigurableApplicationContext
+
+class DelayedShutdownHook(private val applicationContext: ConfigurableApplicationContext) : Thread() {
+    override fun run() {
+        try {
+            // https://github.com/kubernetes/kubernetes/issues/64510
+            // https://nav-it.slack.com/archives/C5KUST8N6/p1543497847341300
+            sleep(5000L)
+        } catch (e: InterruptedException) {
+            e.printStackTrace()
```
Comment (hensol) [inline src/main/kotlin/no/nav/dokgen/configuration/DelayedShutdownHook.kt:12]: better to use a logger.

**Review by hensol — CHANGES_REQUESTED** [review]
Hei,
Har du testet at denne fungerer sammen med malene i familie-ba-dokgen?

**Review by henninghaakonsen — DISMISSED** [review]
I have not looked at the implementation in detail, but it looks good 👍

File: pom.xml (hunk 1 above)
Comment (MengZhuNAV) [reply pom.xml]: We dont actually need it. Removed!

File: pom.xml (hunk 2 above)
Comment (MengZhuNAV) [reply pom.xml]: True, removed!

File: src/main/kotlin/no/nav/dokgen/configuration/DelayedShutdownHook.kt (hunk 3 above)
Comment (MengZhuNAV) [reply src/main/kotlin/no/nav/dokgen/configuration/DelayedShutdownHook.kt]: Agree and changed!

**Review by hensol — APPROVED** [review]

**Pull request merged**